tokenserver/assignment/sqlnode/sql.py

```diff
@@ -181,6 +181,19 @@
 """)
 
 
+# Migrate a user to spanner
+_MIGRATE_USER = sqltext("""\
+update
+    users
+set
+    migration_state = :migration_state,
+    nodeid = :nodeid
+where
+    email = :email
+    and service = :service
+""")
```

Comment:
This is going to have unintended side-effects by affecting historical node-assignment records.

The existing queries in this file affect a node migration by marking the user's current record as "replaced" and then creating a new record for the new node assignment. There is a separate background process that goes through and cleans up all the replaced records, including messaging out to the old storage nodes to make sure they delete obsoleted user data.

This new query would change the nodeid on all those historical records, meaning that background process can't do its job properly, and potentially putting us in violation of GDPR and other privacy legislation (because we can no longer delete the user's historical sync data in response to an explicit deletion request).

Would is suffice to follow the same approach here, adding a new recording and marking the old ones as replaced, without modifying the historical records?

Reply:
Removed update_user